### DISCUSSION TITLE
Pi Zero W works inverse

### DISCUSSION THREAD

**Discussion by dragaus** [gpiozero/gpiozero#970] [Q&A] [answered]
Operating system: Raspbian Buster
Python version: 3.7.3
Pi model: Pi Zero W
GPIO Zero version: 1.6.2
Pin factory used:  RPiGPIO

I'm trying to replicate the code of fade and blink but the led is responding contrary to what is expected for example if im tryng to turn on the led I have to use led.off() or if i used led.value = 0 it turns on, theres some configuration i'm missing

```python
from gpiozero import PWMLED
from time import sleep

led = PWMLED(17)

while True:
    led.value = 0 #sould be off but insted turns on
    sleep(1)
    led.value = 0.5 # half power
    sleep(1)
    led.value = 1 # should be on but insted turns off
    sleep(1)
```

**Comment by bennuttall** [comment, marked as answer]
This depends on your wiring. You should have the long leg to a GPIO and the short leg to GND, then the code above should work. I think if you connect to GPIO and 3V3 instead, it'll be inverted. Alternatively you can use `active_high` parameter to invert it:

```python
led = PWMLED(17, active_high=False)
```

See https://gpiozero.readthedocs.io/en/stable/api_output.html#pwmled

**Reply by lurch** [reply]
Slight mis-formatting (and a typo) in your reply @bennuttall

**Reply by bennuttall** [reply]
whoops - thanks, fixed

**Reply by lurch** [reply]
"You should have the long leg to a GPIO and the short leg to GND" - not forgetting the resistor, of course :slightly_smiling_face:  (it doesn't matter if the resistor goes between LED and GPIO, or between LED and GND)

**Reply by dragaus** [reply]
Thanks thats exactly where I was wrong I was trying to connect with the 3V3 volt Thanks a lot @bennuttall
